runtime/lib/ttmetal/command_queue.cpp

```diff
@@ -426,7 +426,26 @@ void CQExecutor::execute(
 void CQExecutor::execute(
     ::tt::target::metal::EnqueueWriteBufferCommand const *command) {
   ZoneScopedN("EnqueueWriteBufferCommand");
-  throw std::runtime_error("Unsupported EnqueueWriteBufferCommand");
+  assert(buffers.find(command->dst()->global_id()) != buffers.end() &&
+         "Buffer not allocated");
+  auto buffer = buffers[command->dst()->global_id()];
+  constexpr bool blocking = false;
+  switch (command->src_type()) { // currently only supporting ConstantBuffer32
+  case tt::target::metal::HostBuffer::ConstantBuffer32: {
+    auto constantBuffer32 = command->src_as_ConstantBuffer32();
+    assert(constantBuffer32->data() != nullptr &&
+           "Only constant data supported");
+    assert(command->dst()->size() ==
+               buffers[command->dst()->global_id()]->size() &&
+           "Shape mismatch");
+    std::vector<uint32_t> vec(command->dst()->size(),
+                              (*constantBuffer32->data())[0]);
+    ::tt::tt_metal::EnqueueWriteBuffer(*cq, buffer, vec, blocking);
+    break;
+  }
+  default:
+    throw std::runtime_error("Unsupported buffer type");
```

Comment:
Probably good idea to change this to say HostBuffer instead of buffer just to be a tiny bit more clear